src/main/java/org/apache/commons/collections4/bloomfilter/SetOperations.java

```diff
@@ -16,147 +16,172 @@
  */
 package org.apache.commons.collections4.bloomfilter;
 
-import org.apache.commons.collections4.bloomfilter.hasher.Shape;
+import java.util.function.LongBinaryOperator;
+import java.util.function.LongPredicate;
 
 /**
- * Implementations of set operations on Bloom filters.
+ * Implementations of set operations on BitMapProducers.
  *
+ * @since 4.5
  */
 public final class SetOperations {
 
     /**
-     * Calculates the Cosine distance between two Bloom filters.
+     * Calculates the cardinality of the result of a LongBinaryOperator using the
+     * {@code BitMapProducer.makePredicate} method.
+     * @param first the first BitMapProducer
+     * @param second the second BitMapProducer
+     * @param op a long binary operation on where x = {@code first} and y = {@code second} bitmap producers.
+     * @return the calculated cardinality.
+     */
+    private static int cardinality(BitMapProducer first, BitMapProducer second, LongBinaryOperator op) {
+        int[] cardinality = new int[1];
+
```

Comment:
With my updated method this would be:
```Java
        int[] cardinality = new int[1];
        first.forEachBitMapPair(second, (x, y) -> {
            cardinality[0] += Long.bitCount(op.applyAsLong(x, y));
            return true;
        });
        return cardinality[0];
```